checks/imap/imap.go

```diff
@@ -0,0 +1,19 @@
+package imap
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/emersion/go-imap/client"
+)
+
+// New creates a new IMAP health check function.
+func New(client *client.Client) func(ctx context.Context) error {
```

Comment:
This would assume that the user will actually use `github.com/emersion/go-imap` and create a client for us. Should we maybe pass connection parameters instead and create a client ourselves? Otherwise the user will need to get to know how to construct a proper client with this library first, which reduces the usefulness of this health check slightly.

Reply:
Yes you are right. For the user it is easier to pass only the address instead of a client